packages/amplify-graphql-model-transformer/src/resources/model-resource-generator.ts

```diff
@@ -0,0 +1,670 @@
+                `${subscription.typeName}.${subscription.fieldName}.{slotName}.{slotIndex}.req.vtl`,
+              ),
+            );
+          }
+          resolver.mapToStack(context.stackManager.getStackFor(subscription.resolverLogicalId, def!.name.value));
+          context.resolvers.addResolver(subscription.typeName, subscription.fieldName, resolver);
+        });
+      }
+    });
+  }
+
+  generateGetResolver = (
+    ctx: TransformerContextProvider,
+    type: ObjectTypeDefinitionNode,
+    typeName: string,
+    fieldName: string,
+    resolverLogicalId: string,
+  ): TransformerResolverProvider => {
+    const isSyncEnabled = ctx.isProjectUsingDataStore();
+    const dataSource = this.datasourceMap[type.name.value];
+    const resolverKey = `Get${generateResolverKey(typeName, fieldName)}`;
+    const vtlGenerator = this.getVTLGenerator();
+    const requestConfig = {
+      operation: 'GET',
+      operationName: fieldName,
+    };
+    const responseConfig = {
+      ...requestConfig,
+      isSyncEnabled,
+      modelName: type.name.value,
+    };
+    if (!this.resolverMap[resolverKey]) {
+      this.resolverMap[resolverKey] = ctx.resolvers.generateQueryResolver(
+        typeName,
+        fieldName,
+        resolverLogicalId,
+        dataSource,
+        MappingTemplate.s3MappingTemplateFromString(vtlGenerator.generateGetRequestTemplate(requestConfig), `${typeName}.${fieldName}.req.vtl`),
+        MappingTemplate.s3MappingTemplateFromString(vtlGenerator.generateGetResponseTemplate(responseConfig), `${typeName}.${fieldName}.res.vtl`),
+      );
+    }
+    return this.resolverMap[resolverKey];
+  };
+
+  generateListResolver = (
+    ctx: TransformerContextProvider,
+    type: ObjectTypeDefinitionNode,
+    typeName: string,
+    fieldName: string,
+    resolverLogicalId: string,
+  ): TransformerResolverProvider => {
+    const isSyncEnabled = ctx.isProjectUsingDataStore();
+    const dataSource = this.datasourceMap[type.name.value];
+    const resolverKey = `List${generateResolverKey(typeName, fieldName)}`;
+    const vtlGenerator = this.getVTLGenerator();
+    const requestConfig = {
+      operation: 'LIST',
+      operationName: fieldName,
+    };
+    const responseConfig = {
+      ...requestConfig,
+      isSyncEnabled,
+      mutation: false,
+    };
+    if (!this.resolverMap[resolverKey]) {
+      this.resolverMap[resolverKey] = ctx.resolvers.generateQueryResolver(
+        typeName,
+        fieldName,
+        resolverLogicalId,
+        dataSource,
+        MappingTemplate.s3MappingTemplateFromString(vtlGenerator.generateListRequestTemplate(requestConfig), `${typeName}.${fieldName}.req.vtl`),
+        MappingTemplate.s3MappingTemplateFromString(
+          vtlGenerator.generateDefaultResponseMappingTemplate(responseConfig),
+          `${typeName}.${fieldName}.res.vtl`,
+        ),
+      );
+    }
+    return this.resolverMap[resolverKey];
+  };
+
+  generateCreateResolver = (
```

Comment:
Can all these methods be `private`?

Reply:
I don't think we need to bring the `generate*Resolver` into resource provider. Isn't the createTable call alone differ between DDB vs RDS?

Reply:
The problem is that once you start creating the table, the process of creating the table accesses many of the attributes of the transformer plugin. And, creating the table populates an attribute that is used by the resolvers. You could force it so that they are still in their original classes but you get a messy coupling between the two